src/app/appointment/appointment.component.ts

```diff
@@ -50,6 +50,9 @@ export class AppointmentComponent implements OnInit, OnDestroy {
     this.route.params
       .filter(res => res.hasOwnProperty('tab'))
       .subscribe(res => this.currentTab = (<any>res).tab);
+
+    // periodically update countdown
+    setInterval(() => this.setCountdown(), 5000);
```

Comment:
Please don't use `setInterval`. It will run forever - even when the component is destroyed. 

```ts
// in constructor
this.countdownSub = Observable.interval(5000)
  .subscribe(() => this.setCountdown());

// in onDestroy
this.countdownSub.ubsubscribe();
```